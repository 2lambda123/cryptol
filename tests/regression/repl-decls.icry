let x -<- y = x - y; infixl 5 -<-; let (-<-) : Integer -> Integer -> Integer
let x ->- y = x - y; infixr 5 ->-; let (->-) : Integer -> Integer -> Integer

42 -<- 10 -<- 100
42 ->- 10 ->- 100
42 ->- 10 -<- 100

let even x = if x == 0 then True else odd (x-1); let odd x = if x == 0 then False else even (x-1)

:t even
:t odd

even 5
even 4

odd 5
odd 6

let even' : Integer -> Bool; let even' x = if x == 0 then True else odd' (x-1); let odd' x = if x == 0 then False else even' (x-1)

:t even'
:t odd'

even' 5
even' 4

odd' 5
odd' 6

type X n = [n]Integer
take [1...] : X 42

type constraint CS t = (Ring t, Logic t, Integral t)

let f : {a} CS a => a -> Integer; let f x = [1..100]@( (x + fromInteger 5) && fromInteger `0x1f)

:t f
f 0x1234
