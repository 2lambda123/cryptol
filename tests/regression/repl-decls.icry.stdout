Loading module Cryptol
-68
132

[error] at repl-decls.icry:6:4--6:7 and repl-decls.icry:6:11--6:14
    The fixities of
      • (->-) (precedence 5, right-associative)
      • (-<-) (precedence 5, left-associative)
    are not compatible.
    You may use explicit parentheses to disambiguate.
even : {a} (Eq a, Ring a, Literal 1 a) => a -> Bit
odd : {a} (Eq a, Ring a, Literal 1 a) => a -> Bit
False
True
True
False
even' : Integer -> Bool
odd' : Integer -> Bit
False
True
True
False
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36,
 37, 38, 39, 40, 41, 42]
f : {a} (Ring a, Logic a, Integral a) => a -> Integer
26
